README.md
# Office.TXT

officetxt (Office.TXT) - the free writer's command line tool suite


* home  :: [github.com/officetxt/officetxt](https://github.com/officetxt/officetxt)
* bugs  :: [github.com/officetxt/officetxt/issues](https://github.com/officetxt/officetxt/issues)
* gem   :: [rubygems.org/gems/officetxt](https://rubygems.org/gems/officetxt)
* rdoc  :: [rubydoc.info/gems/officetxt](http://rubydoc.info/gems/officetxt)



## Usage

**See the [Office.TXT documentation »](https://officetxt.github.io/)**

Use

```
$ officetxt
-or-
$ txt
```

to listed included / installed tools e.g. resulting in:

```
Welcome to officetxt/0.1.0:

Tool versions installed:
  journaltxt/1.0.1
  jekyll/3.5.0
    jekyll-import/0.12.0
    jekyll-avatar/0.4.2
    jekyll-planet/0.2.1
  mrhyde/0.1.1
  drjekyll/1.0.1
  octopod/0.9.0
  slideshow/3.1.0
  pluto/1.2.3
  rouge/1.11.1
  kramdown/1.13.2
  word-to-markdown/1.1.7
  quik/0.3.0
```


## Install

Just install the gem:

    $ gem install officetxt


## License

![](https://publicdomainworks.github.io/buttons/zero88x31.png)

The `officetxt` scripts are dedicated to the public domain.
Use it as you please with no restrictions whatsoever.

## Questions? Comments?

Post them to the [wwwmake forum](http://groups.google.com/group/wwwmake). Thanks!


HISTORY.md
### 0.0.1 / 2017-07-23

* Everything is new. First release.


test/helper.rb
# encoding: utf-8

## minitest setup
require 'minitest/autorun'


$RUBYLIBS_DEBUG = true

## our own code
require 'officetxt'


test/test_version.rb
# encoding: utf-8

###
#  to run use
#     ruby -I ./lib -I ./test test/test_version.rb


require 'helper'


class TestVersion < MiniTest::Test


  def test_version

    puts Officetxt::VERSION

    puts "  jekyll/#{Jekyll::VERSION}"
    puts "    jekyll-import/#{JekyllImport::VERSION}"
    puts "    jekyll-avatar/#{Jekyll::Avatar::VERSION}"

    puts "    jekyll-planet/#{JekyllPlanet::VERSION}"

=begin
    puts "  jekyll-feed/#{Jekyll::Feed::VERSION}"
   no: version in ruby just in gemspec e.g.
   Gem::Specification.new do |spec|
     spec.name          = "jekyll-feed"
     spec.version       = "0.9.2"

     todo: try to get info from gemspec??
=end


    puts "  octopod/#{Jekyll::Octopod::VERSION::STRING}"


=begin
## fix Octopod

puts "  jekyll-octopod/#{Jekyll::Octopod::VERSION}"

C:/prg/ri/v310/Ruby2.1.0/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:68:in `require': C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:32: void value expression (SyntaxError)
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:33: syntax error, unexpected '<', expecting keyword_end
        <script>
         ^
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:35: syntax error, unexpected '<'
        </script>
         ^
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:37: syntax error, unexpected tSTRING_BEG, expecting keyword_end
...ipts/podigee-podcast-player.js">
...                               ^
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:37: unterminated string meets end of file
=end

    puts "  quik/#{Quik::VERSION}"
    puts    Quik.banner

    puts "  mrhyde/#{MrHyde::VERSION}"
    puts    MrHyde.banner

    puts "  drjekyll/#{DrJekyll::VERSION}"
    puts    DrJekyll.banner

    puts "  slideshow/#{SlideshowCli::VERSION}"   ## note: Slideshow::VERSION is for models
    puts    SlideshowCli.banner

    puts "  journaltxt/#{Journaltxt::VERSION}"
    puts    Journaltxt.banner

    puts "  pluto/#{PlutoCli::VERSION}"   ## note: Pluto::VERSION is for models
    puts    PlutoCli.banner


    puts "  rouge/#{Rouge.version}"           ## incl.  rougify
    puts "  kramdown/#{Kramdown::VERSION}"
    puts "  word-to-markdown/#{WordToMarkdown::VERSION}"


    ###
    ##  ## fix:  hexapdf requires Ruby version >= 2.3.
    ## puts "  hexapdf/#{HexaPDF::VERSION}"

    Officetxt.about


    assert true
    ## assume everything ok if get here
  end

end # class TestVersion


lib/officetxt.rb
# encoding: utf-8


require 'pp'


## add tools
require 'jekyll'
require 'quik'
require 'mrhyde/tools'
require 'drjekyll'
require 'slideshow'
require 'journaltxt'
require 'pluto'

require 'kramdown'
require 'word-to-markdown'
require 'rouge'


require 'jekyll-import'
require 'jekyll-import/version'   ## note: not auto-loaded with gem

## require 'jekyll-octopod'

### add more jekyll plugins plus jekyll-admin
###   plus add some themes!!!

require 'jekyll-avatar'
require 'jekyll-avatar/version'   ## note: not auto-loaded with plugin
require 'jekyll-feed'


require 'jekyll/planet'


require 'octopod/version'   ## note: just auto-load version script


# our own code
require 'officetxt/version'   # note: let version always go first



module Officetxt

def self.about

   puts
   puts
   puts "Welcome to officetxt/#{Officetxt::VERSION}:"
   puts
   puts "Tool versions installed:"

   puts "  journaltxt/#{Journaltxt::VERSION}"
   puts "  jekyll/#{Jekyll::VERSION}"
   puts "    jekyll-import/#{JekyllImport::VERSION}"
   puts "    jekyll-avatar/#{Jekyll::Avatar::VERSION}"
   puts "    jekyll-planet/#{JekyllPlanet::VERSION}"

=begin
    puts "  jekyll-feed/#{Jekyll::Feed::VERSION}"
   no: version in ruby just in gemspec e.g.
   Gem::Specification.new do |spec|
     spec.name          = "jekyll-feed"
     spec.version       = "0.9.2"

     todo: try to get info from gemspec??
=end


=begin
## fix Octopod

puts "  jekyll-octopod/#{Jekyll::Octopod::VERSION}"

C:/prg/ri/v310/Ruby2.1.0/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:68:in `require': C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:32: void value expression (SyntaxError)
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:33: syntax error, unexpected '<', expecting keyword_end
        <script>
         ^
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:35: syntax error, unexpected '<'
        </script>
         ^
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:37: syntax error, unexpected tSTRING_BEG, expecting keyword_end
...ipts/podigee-podcast-player.js">
...                               ^
C:/prg/ri/v310/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/jekyll-octopod-0.9.0/lib/jekyll/podigee_player_tag.rb:37: unterminated string meets end of file
=end

   puts "  mrhyde/#{MrHyde::VERSION}"
   puts "  drjekyll/#{DrJekyll::VERSION}"

   puts "  octopod/#{Jekyll::Octopod::VERSION::STRING}"


   puts "  slideshow/#{SlideshowCli::VERSION}"   ## note: Slideshow::VERSION is for models

   puts "  pluto/#{PlutoCli::VERSION}"   ## note: Pluto::VERSION is for models


   puts "  rouge/#{Rouge.version}"           ## incl.  rougify
   puts "  kramdown/#{Kramdown::VERSION}"
   puts "  word-to-markdown/#{WordToMarkdown::VERSION}"   # incl. w2m

   puts "  quik/#{Quik::VERSION}"


    ### todo:
    ##  (auto-)issue list commands for:   !!!!!!!!!!
    ##  drjekyll
    ##  mrhyde
    ##  slideshow
    ##  pluto
    ##  quik

    ###
    ##  ## fix:  hexapdf requires Ruby version >= 2.3.
    ## puts "  hexapdf/#{HexaPDF::VERSION}"


end  # method about

end # class Officetxt



# say hello
puts Officetxt.banner    if defined?( $RUBYLIBS_DEBUG )
